src/view.c

```diff
@@ -1261,7 +1311,75 @@ view_move_to_edge(struct view *view, enum view_edge direction, bool snap_to_wind
 		}
 	}
 
-	view_move(view, view->pending.x + dx, view->pending.y + dy);
+	if (dx != 0 || dy != 0) {
+		/* Move the window if a change was discovered */
+		view_move(view, view->pending.x + dx, view->pending.y + dy);
+		return;
+	}
+
+	/* If the view is maximized, do not attempt to jump displays */
+	if (view->maximized != VIEW_AXIS_NONE) {
+		return;
+	}
+
+	/* Otherwise, move to edge of next adjacent display, if possible */
+	struct output *output = view_get_adjacent_output(view, direction);
+	if (!output) {
+		return;
+	}
+
+	/* When jumping to next output, attach to edge nearest the motion */
+	struct wlr_box usable = output_usable_area_in_layout_coords(output);
+	struct border margin = ssd_get_margin(view->ssd);
+
+	/* Bounds of the possible placement zone in this output */
+	int left = usable.x + rc.gap + margin.left;
+	int right = usable.x + usable.width - rc.gap - margin.right;
+	int top = usable.y + rc.gap + margin.top;
+	int bottom = usable.y + usable.height - rc.gap - margin.bottom;
+
+	/* Default target position on new output is current target position */
+	dx = view->pending.x;
```

Comment:
At this point we change the meaning of dx,dy from 'delta' (or 'distance') to 'destination'.
Might it be tidier to call them destination_x and destination_y from this point onwards?